### ISSUE TITLE
A false positive for C416

### ISSUE THREAD

**Issue by arnimarj** [adamchainz/flake8-comprehensions#204]
Hi,

I saw some fixes were committed for C416 false positives before, and I believe I've found a new one.

I was working on some code, which converts a list of 2-element sequences to a list of 2-tuples:

```python
from typing import List, Tuple

a_list_of_2lists: List[List[int]] = [[i, i + 1] for i in range(10)]

a_list_of_2tuples: List[Tuple[int, int]] = [(a, b) for a, b in a_list_of_2lists]
```

Incidentally, the code passes if I create a 3-tuple instead:

```python
a_list_of_3tuples: List[Tuple[int, int, str]] = [(a, b, 'foobar') for a, b in a_list_of_2lists]
```

Then I had this idea:
```
a_list_of_2tuples: List[Tuple[int, int]] = [tuple((a, b)) for a, b in a_list_of_2lists]
```
but that gave me a C409

Cheers

**Comment by arnimarj** [comment]
My current fix is to create a 2-tuple factory function:
```python
def _as_tuple(a: _T, b: _TT) -> Tuple[_T, _TT]:
	return (a, b)
```

**Comment by adamchainz** [comment]
C416 currently assumes an iterable using unpacking is an iterable of tuples, which your example proves is not always the case. I'll remove the unpacking path of it.

**Comment by adamchainz** [comment]
Fix released in version 3.1.4: https://pypi.org/project/flake8-comprehensions/3.1.4/

**Comment by arnimarj** [comment]
That's some seriously impressive turnaround time....many thanks!
